include/zephyr/net/prometheus/counter.h

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2024 Mustafa Abdullah Kus, Sparse Technology
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_PROMETHEUS_COUNTER_H_
+#define ZEPHYR_INCLUDE_PROMETHEUS_COUNTER_H_
+
+/**
+ * @file
+ *
+ * @brief Prometheus counter APIs.
+ *
+ * @addtogroup prometheus
+ * @{
+ */
+
+#include <zephyr/sys/iterable_sections.h>
+#include <zephyr/net/prometheus/metric.h>
+
+/**
+ * @brief Type used to represent a Prometheus counter metric.
+ *
+ * * References
+ * * See https://prometheus.io/docs/concepts/metric_types/#counter
+ */
+struct prometheus_counter {
+	/** Base of the Prometheus counter metric */
+	struct prometheus_metric *base;
+	/** Value of the Prometheus counter metric */
+	unsigned long value;
+};
+
+/**
+ * @brief Prometheus Counter definition.
+ *
+ * This macro defines a Counter metric.
+ *
+ * @param _name The channel's name.
+ * @param _detail The metric base.
+ *
+ * @example
```

Comment:
Again not the right tag
You probably just want to use `@code`. Applies elsewhere in the PR

Reply:
See why it's wrong here: https://builds.zephyrproject.io/zephyr/pr/72804/docs/doxygen/html/structprometheus__counter.html